pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -405,6 +409,100 @@ func (c *client) UpdateTrafficConfig(ctx context.Context, fm FunctionManifest, r
 	return nil
 }
 
+func (c *client) updateTagsConfig(ctx context.Context, fm FunctionManifest) error {
+	getFuncInput := &lambda.GetFunctionInput{
+		FunctionName: aws.String(fm.Spec.Name),
+	}
+	output, err := c.client.GetFunctionWithContext(ctx, getFuncInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case lambda.ErrCodeInvalidParameterValueException:
+				return fmt.Errorf("invalid parameter given: %w", err)
+			case lambda.ErrCodeServiceException:
+				return fmt.Errorf("aws lambda service encountered an internal error: %w", err)
+			case lambda.ErrCodeTooManyRequestsException:
+				return fmt.Errorf("request throughput limit was exceeded: %w", err)
+			case lambda.ErrCodeResourceNotFoundException:
+				return fmt.Errorf("resource not found: %w", err)
+			}
+		}
+		return fmt.Errorf("unknown error occurred on list tags of Lambda function %s: %w", fm.Spec.Name, err)
+	}
+
+	functionArn := aws.StringValue(output.Configuration.FunctionArn)
+	currentTags := aws.StringValueMap(output.Tags)
+	// Skip if there are no changes on tags.
+	if reflect.DeepEqual(currentTags, fm.Spec.Tags) {
+		return nil
+	}
+
+	// Skip untag resource if remote resource has no tags.
+	if len(output.Tags) == 0 {
+		goto Tags
```

Comment:
Can you update to avoid using "goto"? (It is fine but for simplicity, we are not using it in the whole source code.)

Reply:
For example, we can break this to smaller functions.

Reply:
This `goto` is the result when I trying to follow the __return early__ and __avoid nested if__ conventions 🙏 
In short, we only need to (and only be able to) remove tags of the remote function in case it's existed. Another option for the implementation of this part is
```golang
         if len(output.Tags) > 0 {
		tagsKeys := make([]string, 0, len(currentTags))
		for k := range currentTags {
			tagsKeys = append(tagsKeys, k)
		}
		untagInput := &lambda.UntagResourceInput{
			Resource: aws.String(functionArn),
			TagKeys:  aws.StringSlice(tagsKeys),
		}
		_, err = c.client.UntagResourceWithContext(ctx, untagInput)
		if err != nil {
			if aerr, ok := err.(awserr.Error); ok {
				switch aerr.Code() {
				case lambda.ErrCodeInvalidParameterValueException:
					return fmt.Errorf("invalid parameter given: %w", err)
				case lambda.ErrCodeServiceException:
					return fmt.Errorf("aws lambda service encountered an internal error: %w", err)
				case lambda.ErrCodeTooManyRequestsException:
					return fmt.Errorf("request throughput limit was exceeded: %w", err)
				case lambda.ErrCodeResourceNotFoundException:
					return fmt.Errorf("resource not found: %w", err)
				case lambda.ErrCodeResourceConflictException:
					return fmt.Errorf("resource already existed or in progress: %w", err)
				}
			}
			return fmt.Errorf("unknown error occurred on untags for Lambda function %s: %w", fm.Spec.Name, err)
		}
	}
```
How do you think about it? 🤔